### ISSUE TITLE
WCAG level A issue on active button

### ISSUE THREAD

**Issue by Shnoulle** [twbs/bootstrap#37365]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Active button use only color for difference with inactive button.

It's break Level A criterion on WCAG, and it make unusuable with forced color mode

### Reduced test cases

- Codepen :  https://codepen.io/team/bootstrap/pen/qBamdLj
- getbootstrap website https://getbootstrap.com/docs/5.2/components/buttons/#toggle-states
- With forced color sample ![Capture d’écran du 2022-10-24 11-38-24](https://user-images.githubusercontent.com/1439428/197498783-f4750aba-33a4-4cb5-b92d-605a7e34f063.png)

(see https://support.mozilla.org/en-US/kb/change-fonts-and-colors-websites-use#w_custom-font-color for firefox settings, i don't found settings for Chrome)

### What operating system(s) are you seeing the problem on?

Linux

### What browser(s) are you seeing the problem on?

Firefox

### What version of Bootstrap are you using?

v5.2.2

**Comment by patrickhlauke** [comment]
@mdo the active class should have a much stronger differentiation with the non-active class. if we're only going by a change of background, the difference between the non-active and active background should be at least a ratio of 3:1 (see the note in )

so, for the case of the blue buttons, `#052A6C` for instance...

![screen of non-active and active toggles, with the latter using the darker background recommended above](https://user-images.githubusercontent.com/895831/197584768-590db655-2450-4bca-aa0f-ecf712c1a570.png)

I'm pretty sure I mentioned this ages ago when we looked at colour contrasts in general. 

as for forced colours, that falls outside of the scope of WCAG itself. we've been discussing separately how to address this (as this will require more thought than just for toggles here) https://github.com/twbs/bootstrap/issues/35941
in the case of active/inactive, one idea would be to add an additional outline, or thicker outline, when active, but we can discuss that in the other issue

**Comment by patrickhlauke** [comment]
incidentally, purely from a use of color perspective (as said, leaving aside the forced colors/high contrast issue), the new styles I introduced recently for the outline toggles https://getbootstrap.com/docs/5.2/forms/checks-radios/#outlined-styles is probably the best, with its distinction between "only border" and "full background" to differentiate regular and active toggles (though doesn't tackle disabled)

**Comment by patrickhlauke** [comment]
this is likely something we want to tackle properly in v6, rather than v5 (together with a good forced color story)

**Comment by Shnoulle** [comment]
> @mdo the active class should have a much stronger differentiation with the non-active class. if we're only going by a change of background, the difference between the non-active and active background should be at least a ratio of 3:1 (see the second note in https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html ... which, incidentally, i wrote)

Thank you:)

A question here about forced-color : it's not included in criterion 1.4.1: Use of Color ?

Have a solid contrast difference are OK ? 

>  It aims to ensure that sighted users who cannot distinguish between some colors can still understand content.

**Comment by patrickhlauke** [comment]
> A question here about forced-color : it's not included in criterion 1.4.1: Use of Color ?

No, 1.4.1 does not cover user adaptations like Windows High Contrast / forced contrast at this stage

> Have a solid contrast difference are OK ?

As per https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html a contrast ratio of 3:1 or higher counts as not just being a "use of color" alone, so yes, solid contrast is ok.

We do plan to make additional adaptations that go beyond WCAG in #35941 but for these toggle buttons, it will likely be about adding an additional border/outline or similar (something that does not rely at all on specific backgrounds/colours)

**Comment by mdo** [comment]
`v6-dev` redoes the button variants. It might not be a full WCAG compatible situation yet, but it does provide a clearer path forward IMO with more explicit controls for button variants and their states.
